### ISSUE TITLE
Categorical Variables

### ISSUE THREAD

**Issue by jnoppenheim** [blei-lab/edward#267]
I apologize in advance if I'm missing something obvious here.  How would one go about setting up a model in Edward, say the Bayesian Linear Regression example, with categorical "X" covariates?  There's good ways of doing this in Tensor Flow and Stan - I'm wondering how we set this all up to run with Edward.

Thanks!

**Comment by dustinvtran** [comment]
This [linear model](https://www.tensorflow.org/versions/r0.11/tutorials/wide/index.html) tutorial from TensorFlow might be helpful. Basically for discrete covariates you can use [`tf.SparseTensor`](https://www.tensorflow.org/versions/r0.11/api_docs/python/sparse_ops.html#SparseTensor).

**Comment by jnoppenheim** [comment]
Right, this all works in TensorFlow with SciKit Learn,  however I'm having issues with Edward.  For instance, if I modify the Bayesian Linear Regression example to change the X variable to a sparse tensor (it works with a regular tensor), Edward's MFVI crashes.

It's also not clear how I could combine individual sparse and non-sparse tensors to pass as a single variable to Edward.

I know the example of linear regression is a little silly, but I'd like to be able to build complex models with categorical data and use Edward's inference capabilities.

Thanks

**Comment by soteri0s** [comment]
In addition, I understand that currently Edward does not provide a "Categorical" model that could be used for modeling purposes in the context of a multi-class classification task. Do you plan on adding this anytime soon? Is there at this time any workaround to this end that I'm missing?

**Comment by dustinvtran** [comment]
there's a Categorical random variable, and also a Multinomial random variable for a one-hot encoding.  we directly support all distributions available in [TensorFlow](https://www.tensorflow.org/versions/r0.11/api_docs/python/contrib.distributions.html).

**Comment by dustinvtran** [comment]
Would be cool to have a tutorial on how to do this.
